bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -738,6 +738,10 @@ PrePrepareMsgCreationResult ReplicaImp::finishAddingRequestsToPrePrepareMsg(PreP
                                                                             uint32_t maxSpaceForReqs,
                                                                             uint32_t requiredRequestsSize,
                                                                             uint32_t requiredRequestsNum) {
+  if (getReplicaConfig().timeServiceEnabled) {
+    prePrepareMsg->setTime(time_service_manager_->getClockTimePoint());
+  }
+
```

Comment:
I am not sure I understand why we add the time here and conditionally later add it ReplicaImp.cpp:879?

Reply:
Cause later in finishAddingRequestsToPrePrepareMsg the diggest is computed, so the timestamp needs to be set before that.